src/module.c

```diff
@@ -1119,6 +1134,7 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
 
     SeriesType = RedisModule_CreateDataType(ctx, "TSDB-TYPE", TS_UNCOMPRESSED_VER, &tm);
     if (SeriesType == NULL) return REDISMODULE_ERR;
+    RTS_GlobalRedisCtx = RedisModule_GetThreadSafeContext(NULL);
```

Comment:
@ashtul don't use a global variable for the context, it should be short lived in the specific context that it is being used.

Reply:
It is a nice tweak we have in RediSearch.
We keep on passing `RedisModuleCtx` from function to function.

Will add instead to the addCtx.

Reply:
I don't think its a good tweak, its true that it might work now in redis but the API assume its a context based resource and should be allocate to the entire run-time of the database.
I also saw that redis graph allocates it only when needed.

Reply:
@ashtul also, please don't add it to the addCtx, i find the huge structs with many argument counter-productive when reading and understanding large pieces of code. since we have 1 place where we need it just create it and pass it to the function.

Reply:
`RedisModule_GetThreadSafeContext` allocates memory. It is not cheap

Reply:
@ashtul 
1. there's one allocation there which is not that bad, plus we are not trying to optimize for OOO inserts.
2. Context object should be short lived, even if it works now its better not the consume them as a long running resource.